message.go

```diff
@@ -32,15 +37,63 @@ func (msg Message) item() messageSetItem {
 
 func (msg Message) message() message {
 	m := message{
-		MagicByte: 1,
-		Key:       msg.Key,
-		Value:     msg.Value,
-		Timestamp: timestamp(msg.Time),
+		MagicByte:  1,
+		Key:        msg.Key,
+		Value:      msg.Value,
+		Timestamp:  timestamp(msg.Time),
+		Attributes: int8(msg.CompressionCodec) & compressionCodecMask,
 	}
 	m.CRC = m.crc32()
 	return m
 }
 
+// Encode encodes the Message using the CompressionCodec.
+func (msg Message) Encode() (Message, error) {
+	if msg.CompressionCodec == CompressionNoneCode {
+		return msg, nil
+	}
+
+	codec, ok := codecs[msg.CompressionCodec]
+	if !ok {
+		return msg, fmt.Errorf("codec %d not imported.", msg.CompressionCodec)
+	}
+
+	var err error
+	msg.Value, err = transform(msg.Value, codec.Encode)
+	return msg, err
+}
+
+// Decode decodes the Message using the CompressionCodec.
+func (msg Message) Decode() (Message, error) {
```

Comment:
Is there a use-case for exporting those methods?

Reply:
mmm I don't have any. I will remove that.

Reply:
This is going to screw up my tests ! :D

Reply:
> This is going to screw up my tests ! :D

You can un-export the method, then put an exported method in a file called `export_test.go` under the same package and your tests will still work.